src/main/java/it/pagopa/afm/marketplacebe/service/BundleService.java

```diff
@@ -74,6 +73,21 @@ public BundleDetails getBundleById(String idBundle, String idPsp) {
         return modelMapper.map(bundle, BundleDetails.class);
     }
 
+    public void deleteBundleByFiscalCode(String fiscalCode, String idBundle) {
+        var ciBundle = ciBundleRepository.findByIdBundleAndCiFiscalCode(idBundle, fiscalCode)
+                .orElseThrow(() -> new AppException(AppError.CI_BUNDLE_NOT_FOUND, idBundle, fiscalCode));
+        var bundle = bundleRepository.findById(idBundle)
```

Comment:
You should execute this query before the previous one (row #77) in order to avoid an unuseful query if idBundle doesn't exist.

I don't know if the validityDateTo of the attributes should set to `now` for coherence.